File: wing.py
# This Python file uses the following encoding: utf-8
import sys

from PySide6.QtWidgets import QApplication, QMainWindow
from Optimisers.WingQuestion import WingQuestion

# Important:
# You need to run the following command to generate the ui_form.py file
#     pyside6-uic form.ui -o ui_form.py, or
#     pyside2-uic form.ui -o ui_form.py

from ui_wing import Ui_wWing


class WingWindow(QMainWindow):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.ui = Ui_wWing()
        self.ui.setupUi(self)

        self.ui.btnCalculate.clicked.connect(self.btnCalculate_clicked)
        self.ui.btnBack.clicked.connect(self.btnBack_clicked)
        print("Initialised")

    def btnCalculate_clicked(self):
        print("Clicked")
        ShowPlot = False
        if self.ui.cbPlot.isChecked():
            ShowPlot = True

        cl_coeff = self.ui.txtCLCoeff.text()
        cl_offset = self.ui.txtCLOff.text()
        cd_const = self.ui.txtCDConst.text()
        cd_quad = self.ui.txtCDQuad.text()
        initial_guess_alpha = self.ui.txtAlphaGuess.text()
        stall_limit = self.ui.txtStallLimit.text()
        V_cruise = self.ui.txtCruise.text()
        V_landing = self.ui.txtLanding.text()

        self.ui.txtOutput.setText(str(WingQuestion.optimise_wing(cl_coeff, cl_offset, cd_const, cd_quad, initial_guess_alpha, stall_limit, V_cruise, V_landing, ShowPlot)))

    def btnBack_clicked(self):
        from menu import MainWindow
        print("Back")
        self.hide()
        self.ui_menu = MainWindow()
        self.ui_menu.show()


if __name__ == "__main__":
    app = QApplication(sys.argv)
    widget = WingWindow()
    widget.show()
    sys.exit(app.exec())

File: wing.cpp
// wing.cpp
#include "wing.h"
#include "menu.h"
#include "python_utils.h"
#include <QDebug>

WingWindow::WingWindow(QWidget *parent)
    : QMainWindow(parent)
    , ui(new Ui::wWing)
{
    ui->setupUi(this);
    connect(ui->btnCalculate, &QPushButton::clicked,
            this, &WingWindow::btnCalculate_clicked);
    connect(ui->btnBack,      &QPushButton::clicked,
            this, &WingWindow::btnBack_clicked);
    qDebug() << "WingWindow initialized";
}

WingWindow::~WingWindow()
{
    delete ui;
}

void WingWindow::btnCalculate_clicked()
{
    qDebug() << "Calculate clicked";

    //bool showPlot = ui->cbPlot->isChecked();
    QString cl_coeff            = ui->txtCLCoeff->text();
    QString cl_offset           = ui->txtCLOff->text();
    QString cd_const            = ui->txtCDConst->text();
    QString cd_quad             = ui->txtCDQuad->text();
    QString initial_guess_alpha = ui->txtAlphaGuess->text();
    QString stall_limit         = ui->txtStallLimit->text();
    QString V_cruise            = ui->txtCruise->text();
    QString V_landing           = ui->txtLanding->text();

    QStringList args = {
        cl_coeff,
        cl_offset,
        cd_const,
        cd_quad,
        initial_guess_alpha,
        stall_limit,
        V_cruise,
        V_landing,
        //showPlot ? "1" : "0"
    };

    // Call the Python function:
    // Note: assumes a Python wrapper so that
    // module “Optimisers.WingQuestion.WingQuestion”
    // exposes optimise_wing(...)
    QString result = callPythonFunction(
        "Optimisers.WingQuestion.WingQuestion",
        "optimise_wing",
        args
        );

    qDebug() << "Python returned:" << result;
    ui->txtOutput->setText(result);
}

void WingWindow::btnBack_clicked()
{
    qDebug() << "Back clicked";
    hide();
    Menu *m = new Menu(this);
    m->show();
}
